### ISSUE TITLE
Incremental task execution cross-compilation doesn't work

### ISSUE THREAD

**Issue by bmuschko** [gogradle/gogradle#190]
Running the the `build` task subsequently without changing the configuration or code does not make the cross-compilation tasks up-to-date. Each cross-compilation task should be marked up-to-date if the binary file already exists.

**Build execution 1:**

```
$ gradle build --console=plain
Parallel execution is an incubating feature.
Found go 1.8.3 in /usr/local/go/bin/go, use it.
:prepare
Found global GOPATH: /Users/bmuschko/.gopath.
:resolveBuildDependencies
Resolving github.com/mvdan/xurls: commit='d315b61cf6727664f310fa87b3197e9faf2a8513', urls=[https://github.com/mvdan/xurls.git, git@github.com:mvdan/xurls.git]
:installDependencies
:buildDarwinAmd64
:buildNetbsd386
:buildNetbsdAmd64
```

**Build execution 2:**

```
$ gradle build --console=plain
Parallel execution is an incubating feature.
Found go 1.8.3 in /usr/local/go/bin/go, use it.
:prepare
Found global GOPATH: /Users/bmuschko/.gopath.
:resolveBuildDependencies UP-TO-DATE
:installDependencies
:buildDarwinAmd64
:buildNetbsd386
:buildNetbsdAmd64
:build
```

**Comment by a-titorenko** [comment]
see same behavior.

**Comment by blindpirate** [comment]
@bmuschko @a-titorenko Thanks for reporting. This will be fixed in next release `0.9.0`.

**Comment by a-titorenko** [comment]
Thank you!

Let me just check my understanding. As I can see, ‘installDependencies’, despite its name, actually downloads dependencies into ‘vendor’ directory. I think that if they are downloaded and locked, they should not be refreshed from remote even if I chang my code or re-building after cleanup. Also, it should not refresh dependencies if their versions are explicitly specified in the build file. Otherwise, there is not way to get reproducible build.

I’m not sure if it should or shuldn’t update verndored dependencies each time if they are not locked and versions are not specified.

**Comment by blindpirate** [comment]
@a-titorenko Sorry for the late response, I was busy in past 2 days.

There're 2 stages of dependency resolution:

- [Resolve dependencies](https://github.com/gogradle/gogradle/blob/8e26e3e475f85e370989e7db25f0a6d0d387ed00/src/main/java/com/github/blindpirate/gogradle/task/ResolveDependencies.java#L57) is converting dependency declaration into concrete dependency version (and maybe resolve version conflicts), e.g. `github.com/my/package` -> `github.com/my/package @ commit 1a2b3c4d`

- [Install dependencies](https://github.com/gogradle/gogradle/blob/5a0fe42c5714122084e6236c704edfdd22b68001/src/main/java/com/github/blindpirate/gogradle/task/GoInstall.java#L37) is installing concrete dependency into vendor, e.g. downloading/copying to vendor directory. 

Yes, if it's locked they won't be refreshed from remote.

**Comment by blindpirate** [comment]
Closed by `0.9`. Thanks for reporting.
